nexus-tweaks.user.js

```diff
@@ -1144,6 +1144,264 @@ promiseList.push((async () => {
+                "The Rise of Kafa-El", "The Voice", "Under The Boot", "Untouched Wilderness", "Well of Truth", "What Once Was Lost"
+            ],
+            "Breath 3" : [
+                "Bloodlines", "Common Touch", "Cromahl-Hult", "Deluge", "Effervescence", "Eresius Rest", "Fatigued", "Favored of Baraas", "Fire and Ice", "Flakes of Delight",
+                "Fought the Law", "Four Winds", "Harmonious Joining", "Hole in None", "Immutable Law", "Landfall", "Netherland", "Paradise Lost", "Riptide", "Shades of Black",
+                "Shades of Grey", "Sincerest Flattery", "Tainted Fruit", "The Crypt of Maeval", "View from the Top"
+            ],
+            "Breath 2" : [
+                "A Whole New World", "All Ye All Ye Outs In Free", "Beginning of the End", "Blinded by Lust", "Bloodlust", "Bridge Magic Built", "Captain Bloodbeards Folly",
+                "Crystal Farm", "Deck of Cards", "Goldlust", "Ground Zero", "Impassable Dream", "Last Ship Leaving", "Mene Mene Tekel Upharsin", "Smiles All Around", "Snicker-Snack",
+                "Time Sink", "Tree Hugger", "Viva La Revolucion", "Walking Home", "Watery Grave", "When The Walls Came Tumbling Down"
+            ],
+            "Breath 1" : [
+                "Altar of the Ancients", "Big Bang", "Confluence", "Founding Father", "Great Bowl", "Hermits Repose", "Iron Barracks", "Lord Mayor", "Shrine of the Wound", "Slowpoke"
+            ]
+        },
+        "Monster Hunting" : {
+            "Crystal Guardian" : ["Shardbreaker", "Crystal Smasher", "Entropy Incarnate"],
+            "Great Wyrm" : ["Wyrm Hunter", "Wyrm Slayer", "Wyrmsbane"],
+            "Maeval" : ["Stalker of Death", "Hunter of Death", "Foe of Death"]
+        }
+    }
+
+    function classifyBadges(badgesText) {
+        const badgeTexts = badgesText.trim().split(",").map(str => str.trim())
+        let classifiedBadges = {}
+
+        for (let bT of badgeTexts) {
+            let found = false
+            for (let firstLevel in Badges) {
+                for (let secondLevel in Badges[firstLevel]) {
+                    if (Badges[firstLevel][secondLevel].includes(bT)) {
+                        if (!(firstLevel in classifiedBadges)) classifiedBadges[firstLevel] = {}
+                        if (!(secondLevel in classifiedBadges[firstLevel])) classifiedBadges[firstLevel][secondLevel] = []
+                        classifiedBadges[firstLevel][secondLevel].push(bT)
+                        found = true
+                        continue
+                    }
+                }
+                if (found) continue
+            }
+            if (!found) {
+                if (!classifiedBadges.Others) classifiedBadges.Others = []
+                classifiedBadges.Others.push(bT)
+            }
+        }
+
+        return classifiedBadges
+    }
+
+    function printClassifiedBadges(classifiedBadges, badgesNode) {
+        function L1(text, parent) {
```

Comment:
I *think* function definitions like this get hoisted, at least that's my old understanding of JS.
Hence why so many things are written in fat-arrow notation, to keep them scoped locally.
Do you know if that is still the case?